src/cmake-tools.ts

```diff
@@ -123,6 +141,127 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
   get onStatusMessageChanged() { return this._statusMessage.changeEvent; }
   private readonly _statusMessage = new Property<string>(localize('initializing', 'Initializing'));
 
+  /**
+   * Minimum cmake version supported. Currently only used for presets
+   */
+  get minCMakeVersion() { return this._minCMakeVersion; }
+  set minCMakeVersion(version: Version | undefined) { this._minCMakeVersion = version; }
+  private _minCMakeVersion?: Version;
+
+
+  /**
+   * Currently selected configure preset
+   */
+  get configurePreset() { return this._configurePreset; }
+  private _configurePreset: preset.ConfigurePreset | null = null;
+  async setConfigurePreset(configurePreset: string | null) {
+    if (configurePreset) {
+      log.debug(localize('resolving.config.preset', 'Resolving the selected configure preset'));
+      this._configurePreset = await preset.expandConfigurePreset(configurePreset,
+                                                                 lightNormalizePath(this.folder.uri.fsPath || '.'),
+                                                                 await this.sourceDir,
+                                                                 true);
+      if (!this._configurePreset) {
+        log.error(localize('failed.resolve.config.preset', 'Failed to resolve configure preset: {0}', configurePreset));
+        return;
+      }
+      log.debug(localize('loading.new.config.preset', 'Loading new configure preset into CMake driver'));
+      const drv = await this._cmakeDriver;  // Use only an existing driver, do not create one
+      if (drv) {
+        try {
+          this._statusMessage.set(localize('reloading.status', 'Reloading...'));
+          await drv.setConfigurePreset(this._configurePreset);
+          this.workspaceContext.state.configurePresetName = configurePreset;
+          this._statusMessage.set(localize('ready.status', 'Ready'));
+        } catch (error) {
+          vscode.window.showErrorMessage(localize('unable.to.set.config.preset', 'Unable to set configure preset "{0}".', error));
+          this._statusMessage.set(localize('error.on.switch.config.preset', 'Error on switch of configure preset ({0})', error.message));
```

Comment:
{0} in between " is intentional? I get the popup message with {0} as text and not resolved with error.

Reply:
and it's in several other places, just check for pattern "{

Reply:
This code is actually copied from kit (cmekt-tools.ts, line 732). How did you get that popup message?

Reply:
By connecting the preset to a fake artificial kit I had tested for something else (and forgot that it was not valid when I selected it). It had only "name" in it. This will not be a true scenario but here are all the issues I found through this, which I believe should be fixed because who knows how else the following may be reachable via a real scenario.

1. add a new preset --> scan for compilers --> select invalid kit --> name the new preset --> select again the new preset (by the way, I felt a bit confused that by "add a new preset" I didn't automatically select it and I have to press again the button in the status bar) --> the message
2. add a new valid preset and corresponding build/test created from it --> change the configure preset to the invalid one above --> no more message (inconsistent with the above, but message will appear after any future reload) --> build/test presets show unset in status bar but from output channel I think they're set (incompatibility message). Also if I select the original good configure preset, the build/test presets magically change from unset to what they were before.
3. Continuing (2.), a reset state now is working but if I reset state after a reload, it will never do a good reset, it will pop up the configure preset quickPick: if I cancel --> reset does reload but from output channel I deduce that it didn't reset, if I select a good configure preset now then it will properly reset all the state
4. Continuing (2.): after a few more steps which I can't reproduce anymore, any reload will show the bottom right {0} message (expected, configure preset can't be set, it's invalid) but the quickpick isn't automatically shown as I was used to see it before every time a configure preset is invalid (can't be set) or there isn't one at all yet.
5. I have several presets tied to similar invalid kits and after playing with them (set, unset, etc...) I go through various inconsistent states: sometimes when I set such preset I don't see a bottom right message at all (unless I reload), or I see the popup but UI button takes the name of the new invalid preset.... or I see the popup but the button is not changing from "No configure preset selected".